package.json

```diff
@@ -12,6 +12,7 @@
     "test-08:00": "TZ=America/Los_Angeles vitest --run ./src/__tests__/date-utils --coverage=false",
     "test+08:00": "TZ=Asia/Brunei vitest --run ./src/__tests__/date-utils --coverage=false",
     "test-local": "vitest --run",
+    "pretest": "tsc -p tsconfig.unit.json",
```

Comment:
Regression happened when migrating to vitest: https://github.com/cloudscape-design/collection-hooks/pull/73

Unlike ts-jest, vitest does not do type checks when running. Added this step and fixed a bunch of type errors existed in the project